### DISCUSSION TITLE
More options for CD identification

### DISCUSSION THREAD

**Discussion by ben-barnard** [automatic-ripping-machine/automatic-ripping-machine#947] [Ideas]
I have had multiple cases where I'm ripping a CD that can't be identified - so I use the handy search functionality to look it up, and it works great. However, there are some weird cases where the search absolutely refuses to pull the title even when I can definitely identify it on Musicbrainz.  Most recent example:

https://musicbrainz.org/release/ac0660ee-1f07-4ec5-8795-2874720aa232

![image](https://github.com/automatic-ripping-machine/automatic-ripping-machine/assets/34529890/dac718fa-ae5e-49bd-8b15-306d46f4e833)

Even though I have found the correct match, the ARM title update search cannot find it. Could we get another form element here that we can manually provide the ID#, so in this case, ac0660ee-1f07-4ec5-8795-2874720aa232 ?

**Comment by jkesselm** [comment]
I have a number of not-listed disks, some from fairly major publishers such as Harmonia Mundi and Virgin, some from obscure small labels, some one-off demo disks. (You would think that Joan Baez' _Play Me Backward_ would be in the database; apparently not!)

ARM seems to be (over)writing all of these into the same path and filenames, as (eg) `./cds/Unknown Artist/Unknown Album/1 - Track 1.flac`

If would be Really Nice if ARM could instead fall back on asking the user (through the web interface) to provide at least the metadata needed for meaningful filenames/paths.  Suggested user interface: On unknown drive, do not eject at end of rip. Retaining the disk will then as an indication to the user that ARM needs assistance indexing. User can go to the ARM server, select an "Unrecognized Disk" page, and get an editor page which lets them decide to either enter/save the information or cancel the rip; either choice would then eject the disk and let the user proceed with further rips on that drive.

Failing that, at least add a disambiguation field to `Unknown Artist/Unknown Album` -- count, timestamp, disk hash -- so those don't overwrite each other. Users may be able to recognize the disk and  the metadata later by auditioning a track or two.

**Reply by jkesselm** [reply]
Related: I have the Beatles boxed/remastered set. Many of the disks in that collection currently come through as the boxed set (The Beatles/The Beatles)  rather than with their correct album name. I can rename directories and the .flac.m3u file manually, but it feels like this should be fixable from within ARM as we see it happening (and I'm not sure that the renames are enough to fix ARM's database).

Just helping to define the wishlist item, not demanding fixes yesterday!

**Reply by jkesselm** [reply]
Oddity noticed: Sometimes when ARM can't resolve a "The Beatles" disk's identity, the disk- loaded bubble on my Ubuntu Studio machine has just shown me the correct title. Either they're looking in a different database, or they're finding the name somewhere on the disk itself. Might be worth finding out which.

**Reply by jkesselm** [reply]
Another oddity.
Album: "First Contact" by the Black Book Band
Musicbrainz webUI search finds it.
ARM does not find it automagically.
ARM search for album title does not find it.

Want logs?

**Reply by microtechno9000** [reply]
yes please

**Comment by jkesselm** [comment]
Updated the Docker Container, waited for it to stabilize, then attempted again  to rip **First Contact**, by **The Black Book Band**, which does have [a musicbrainz entry.](https://musicbrainz.org/artist/f9a129ef-58c5-42c4-93ef-d9c81250e6f3)

Logfile attached.
[music_cd_171912430009.log](https://github.com/user-attachments/files/15942870/music_cd_171912430009.log)

Two things noted in the log:

1) The attempt to mount the disk into the filesystem failed, though tracks did get ripped. Would that explain the missing metadata? Is there additional diagnostic info I can fetch for you? (It's been a LONG time since I mucked with CD formats.)

2) The album is expected to have 14 tracks. This run found a 15th, though it's only about 1/3 MB rather than the 20+ MB for the actual songs.

This _is_ a relatively early disk from a small label (Dodeka Records), so it's possible that they did something unusual. But I've been finding that roughly 10% of my CDs come up as Unknown, some of them from fairly prestigious labels, so whatever's "unusual" is apparently not uncommon.

FWIW, I tried explicitly looking up **First Contact** through ARM's search. That said it couldn't find the data either.

Very willing to send other data or retest with debugging turned on; just tell me what you need me to do.

(Wish I had the cycles to offer to get involved in the code, but I'm already overcommitted. )

**Reply by jkesselm** [reply]
(I can give you logfiles for others not resolved but found in Musicbranz, if that'd be helpful. But I do recognize that my collection is... quirky.)

**Reply by microtechno9000** [reply]
For music, ARM uses abcde and it grabs details from musicbrainz.

There are many configurations possible with abcde that use alternat sources. I havent used any but understand there are.
